tensorflow/compiler/tf2tensorrt/convert/ops/slice_ops.cc

```diff
@@ -152,6 +176,36 @@ Status HandleDynamicStridedSliceInput(
   nvinfer1::ITensor* input_tensor = slice_layer->getInput(0);
   TRT_ENSURE(input_tensor);
 
+  // When begin_mask or end_mask are set, we have to disregard the begin_tensor
+  // and end_tensor values. In static indices cases, ValidateStridedSliceOp
+  // returns the correct begin_tensor and end_tensor values, however with
+  // dynamic indices the correct shape has to be computed.
+
+  VLOG(2) << "begin_dims before: " <<  DebugString(begin_dims);
+  VLOG(2) << "end_dims before: " <<  DebugString(end_dims);
+  const auto begin_mask = std::bitset<32>(strided_slice_spec.begin_dense_mask);
+  const auto end_mask = std::bitset<32>(strided_slice_spec.end_dense_mask);
+  const auto shrink_axis_mask = std::bitset<32>(strided_slice_spec.shrink_axis_dense_mask);
+  nvinfer1::Dims dims = input_tensor->getDimensions();
+
+  for (int idx = 0; idx < dims.nbDims; ++idx){
+    // VLOG(2) << "begin_mask[" << idx << "]: "<< begin_mask[idx];
+    // VLOG(2) << "end_mask[" << idx << "]: "<< end_mask[idx];
+    // VLOG(2) << "shrink_mask[" << idx << "]: "<< shrink_axis_mask[idx];
```

Comment:
What do you want to do with these lines?

Reply:
These are optional debugging statements. I wanted to keep them commented because slice_ops.cc seems to be a place where we debug for many operations like split, unpack, slice, strided slice etc. and these lines with mask properties just make it simpler to debug. I can remove them if they are better removed.

Reply:
It is either in or out, we can keep debug code for this. You may want to make them VLOG(3).

Reply:
Done